src/api/resources/sample_units/sample_unit_methods.py

```diff
@@ -43,6 +43,11 @@ class SearchNonFieldFilter(django_filters.Filter):
     def filter(self, qs, value):
         value = value or ""
 
+        try:
+            re.compile(value)
+        except re.error:
+            raise exceptions.ValidationError("Invalid search")
```

Comment:
So this exception will be handled and passed to response, and not cause a 500?

Reply:
Yeah causes a 400 error now